scala-package/macros/src/main/scala/org/apache/mxnet/SymbolMacro.scala

```diff
@@ -136,20 +98,67 @@ private[mxnet] object SymbolImplMacros {
     result
   }
 
+  // Convert C++ Types to Scala Types
+  private def typeConversion(in : String, argType : String = "") : String = {
+    in match {
+      case "Shape(tuple)" | "ShapeorNone" => "Shape"
+      case "Symbol" | "NDArray" | "NDArray-or-Symbol" => "Symbol"
+      case "Symbol[]" | "NDArray[]" | "NDArray-or-Symbol[]" | "SymbolorSymbol[]" => "Array[Symbol]"
+      case "float" | "real_t" => "MXFloat"
+      case "int" | "intorNone" | "int(non-negative)" => "Int"
+      case "long" | "long(non-negative)" => "Long"
+      case "double" => "Double"
+      case "string" => "String"
+      case "boolean" => "Boolean"
+      case "tupleof<float>" => "Any"
+      case default => throw new IllegalArgumentException(
+        s"Invalid type for args: $default, $argType")
+    }
+  }
+
+  private def argumentCleaner(argType : String) : (String, Boolean) = {
+    val spaceRemoved = argType.replaceAll("\\s+", "")
+    var commaRemoved : Array[String] = new Array[String](0)
+    // Deal with the case e.g: stype : {'csr', 'default', 'row_sparse'}
+    if (spaceRemoved.charAt(0)== '{') {
+      val endIdx = spaceRemoved.indexOf('}')
+      commaRemoved = spaceRemoved.substring(endIdx + 1).split(",")
+      // commaRemoved(0) = spaceRemoved.substring(0, endIdx+1)
+      commaRemoved(0) = "string"
```

Comment:
could you explain more about the process logic here? I don't quite get the point.

Reply:
The input can be in the format:
e.g: stype : {'csr', 'default', 'row_sparse'}
In which case we need to get rid of the "{}" and set the type as string. This part is just a part of the data cleaning